templates/js-langchain/src/main.js

```diff
@@ -0,0 +1,98 @@
+import { Actor } from 'apify';
+import { ApifyDatasetLoader } from 'langchain/document_loaders/web/apify_dataset';
+import { Document } from 'langchain/document';
+import { HNSWLib } from 'langchain/vectorstores/hnswlib';
+import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
+import { RetrievalQAChain } from 'langchain/chains';
+import { OpenAI } from 'langchain/llms/openai';
+import { rm } from 'node:fs/promises';
+
+import { retrieveVectorIndex, cacheVectorIndex } from './vector_index_cache.js';
+
+await Actor.init();
+
+// There are 2 steps you need to proceed first in order to be able to run this template:
+// 1. If you are running template locally then you need to authenticate to Apify platform by calling `apify login` in your terminal. Without this, you won't be able to run the required Website Content Crawler Actor to gather the data.
+// 2. Configure the OPENAI_API_KEY environment variable (https://docs.apify.com/cli/docs/vars#set-up-environment-variables-in-apify-console) with your OpenAI API key you obtain at https://platform.openai.com/account/api-keys.
+const { OPENAI_API_KEY, APIFY_TOKEN } = process.env;
+
+// You can configure the input for the Actor in the Apify UI when running on the Apify platform or editing storage/key_value_stores/default/INPUT.json when running locally.
+const {
+    startUrls = [{ url: 'https://wikipedia.com' }],
+    maxCrawlPages = 3,
+    forceRecrawl = false, // Enforce a re-crawl of website content and re-creation of the vector index.
+    query = 'What is Wikipedia?',
+    openAIApiKey = OPENAI_API_KEY, // This is a fallback to the OPENAI_API_KEY environment variable when value is not present in the input.
+} = await Actor.getInput() || {};
+
+// Local directory where the vector index will be stored.
+const VECTOR_INDEX_PATH = './vector_index';
+
+if (!openAIApiKey || !openAIApiKey.length) throw new Error('Please configure the OPENAI_API_KEY as environment variable or enter it into the input!');
+if (!APIFY_TOKEN || !APIFY_TOKEN.length) throw new Error('Please configure the APIFY_TOKEN environment variable! Call `apify login` in your terminal to authenticate.');
```

Comment:
Empty strings are `false`-y in JavaScript already, no need to check the `length`
```suggestion
if (!openAIApiKey) throw new Error('Please configure the OPENAI_API_KEY as environment variable or enter it into the input!');
if (!APIFY_TOKEN) throw new Error('Please configure the APIFY_TOKEN environment variable! Call `apify login` in your terminal to authenticate.');
```